chains/manager.go

```diff
@@ -1363,3 +1378,12 @@ func (m *manager) getChainConfig(id ids.ID) (ChainConfig, error) {
 
 	return ChainConfig{}, nil
 }
+
+type snowmanVM struct {
+	block.ChainVM
+	proposerVM *proposervm.VM
+}
+
+func (vm *snowmanVM) GetInitialPreference() ids.ID {
+	return vm.proposerVM.GetInitialPreference()
+}
```

Comment:
We can treat this like an optional interface to make the code in the `chains/manager.go` file simpler but the tradeoff will be that `Transitive` will now have to handle the case that `GetInitialPreference` is not implemented in which case we use the last accepted block. This is not complex to handle, but it just looks a little strange since proposerVM is our only consensus VM so we're making an interface optional even though its only implementation does implement it. This does definitely make the diff in the upstream code/tests cleaner so I'm fine implementing it as an optional interface for now.